charms/kfp-schedwf/tests/unit/test_operator.py

```diff
@@ -1,58 +1,116 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-from contextlib import nullcontext as does_not_raise
+from unittest.mock import MagicMock
 
-import ops
 import pytest
-import yaml
-from oci_image import MissingResourceError
 from ops.model import ActiveStatus, WaitingStatus
 from ops.testing import Harness
 
 from charm import KfpSchedwf
 
-ops.testing.SIMULATE_CAN_CONNECT = True
 
-
-def test_not_leader(harness):
+def test_not_leader(
+    harness,
+    mocked_lightkube_client,
+):
+    """Test that charm waits for leadership."""
     harness.begin_with_initial_hooks()
-    assert harness.charm.model.unit.status == WaitingStatus("Waiting for leadership")
+    assert harness.charm.model.unit.status == WaitingStatus(
+        "[leadership-gate] Waiting for leadership"
+    )
 
 
-def test_image_fetch(harness, oci_resource_data):
+def test_kubernetes_component_created(harness, mocked_lightkube_client):
+    """Test that Kubernetes component is created when we have leadership."""
+    # Needed because the kubernetes component will only apply to k8s if we are the leader
+    harness.set_leader(True)
     harness.begin()
-    with pytest.raises(MissingResourceError):
-        harness.charm.image.fetch()
 
-    harness.add_oci_resource(**oci_resource_data)
-    with does_not_raise():
-        harness.charm.image.fetch()
+    # Need to mock the leadership-gate to be active, and the kubernetes auth component so that it
+    # sees the expected resources when calling _get_missing_kubernetes_resources
+    kubernetes_resources = harness.charm.kubernetes_resources
+    kubernetes_resources.component._get_missing_kubernetes_resources = MagicMock(return_value=[])
 
+    harness.charm.on.install.emit()
 
-def test_install_with_all_inputs(harness, oci_resource_data):
-    harness.set_leader()
-    harness.add_oci_resource(**oci_resource_data)
+    assert isinstance(harness.charm.kubernetes_resources.status, ActiveStatus)
 
-    harness.begin_with_initial_hooks()
+    # Assert that expected amount of apply calls were made
+    # This simulates the Kubernetes resources being created
+    # FIXME: FAILED tests/unit/test_operator.py::test_kubernetes_component_created -
```

Comment:
Do we need to include this comment in the final code?

Reply:
Will remove it

Reply:
Done